include/aspect/particle/output/ascii.h

```diff
@@ -0,0 +1,81 @@
+/*
+ Copyright (C) 2011 - 2015 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef __aspect__particle_output_ascii_h
+#define __aspect__particle_output_ascii_h
+
+#include <aspect/particle/output/interface.h>
+#include <aspect/simulator_access.h>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Output
+    {
+      template <int dim>
+      class ASCIIOutput : public Interface<dim>,
+        public SimulatorAccess<dim>
+      {
+        public:
+          /**
+           * Constructor.
+           */
+          ASCIIOutput();
+
+          /**
+           * Write data about the particles specified in the first argument
+           * to a file. If possible, encode the current simulation time
+           * into this file using the data provided in the second argument.
+           *
+           * @param[in] particles The set of particles to generate a graphical
+           *   representation for
+           * @param [in] data_names The names of the particle properties that
+           *   will be written.
+           * @param [in] data_components The number of property components.
+           *   Equals one for scalar properties and dim for vector properties,
+           *   but any other number is valid as well (e.g. number of compositional
+           *   fields).
```

Comment:
This parameter does not appear to exist any longer. But instead `lengths` now exists for which I can not imagine what it does -- which might suggest that a different name for the variable might also be helpful.